lib/SILOptimizer/Differentiation/PullbackCloner.cpp

```diff
@@ -2515,12 +2515,11 @@ void PullbackCloner::Implementation::visitSILBasicBlock(SILBasicBlock *bb) {
 
     // Get predecessor terminator operands.
     SmallVector<std::pair<SILBasicBlock *, SILValue>, 4> incomingValues;
-    bbArg->getSingleTerminatorOperands(incomingValues);
-
-    // Returns true if the given terminator instruction is a `switch_enum` on
-    // an `Optional`-typed value. `switch_enum` instructions require
-    // special-case adjoint value propagation for the operand.
-    auto isSwitchEnumInstOnOptional =
+    if (bbArg->getSingleTerminatorOperands(incomingValues)) {
```

Comment:
Should we do an early return here to reduce nesting?

Reply:
It seems we'd need to eventually handle `else` case since it at least covers `try_apply` instruction among the others. The fix for activity calculation was pretty straightforward, but I will need to think how to proceed with adjoint propagation. I do not have a testcase for this right now, but hopefully @BradLarson will be able to distill some from their codebase.

For now we just do not silently generate incorrect code :)